target.go

```diff
@@ -111,26 +129,52 @@ func (t *Target) check(ctx context.Context) *Target {
 		logger.Errorw("failed to check if target is known by indexer", "err", err)
 	}
 
-	// Check if the target is an index provider on the expected topic.
-	var ok bool
-	ok, t.Topic, t.HeadProtocol, t.Err = t.hf.supportsHeadProtocol(ctx, t.AddrInfo)
-	switch {
-	case t.Err != nil:
+	if t.Err = t.hf.h.Connect(ctx, *t.AddrInfo); t.Err != nil {
 		t.Status = StatusUnreachable
 		return t
-	case !ok:
-		t.Status = StatusUnindexed
+	}
+	if t.Protocols, t.Err = t.hf.h.Peerstore().GetProtocols(t.AddrInfo.ID); t.Err != nil {
+		t.Status = StatusUnreachable
 		return t
-	default:
-		if t.Topic != t.hf.topic {
-			t.Status = StatusTopicMismatch
-			return t
+	}
+
+	// Get the remote peer agent version, but proceed with other checks if we fail to get it.
+	var ok bool
+	if anyAgentVersion, err := t.hf.h.Peerstore().Get(t.AddrInfo.ID, "AgentVersion"); err != nil {
+		logger.Warnw("Failed to get agent version", "err", err)
+	} else if t.AgentVersion, ok = anyAgentVersion.(string); ok {
+		t.AgentShortVersion = agentShortVersionPattern.ReplaceAllString(t.AgentVersion, "$1")
+	} else if !ok {
+		logger.Warnw("Non-string agent version", "agentVersion", anyAgentVersion)
+	}
+
+	if t.supportsProtocolID(transportsProtocolID) {
+		// Do not populate t.Err with the error that may be returned by queryTransports.
+		// Instead, silently proceed to IPNI related head checking, etc.
+		if t.Transports, err = t.hf.queryTransports(ctx, t.AddrInfo.ID); err != nil {
```

Comment:
should this be in a sub-ctx with a timeout, since it'll block on connecting to the market endpoint of the provider?